frontend/src/stimulus/controllers/dynamic/project-storage-form.controller.ts

```diff
@@ -77,17 +91,24 @@ export default class ProjectStorageFormController extends Controller {
   declare readonly hasProjectFolderSectionTarget:boolean;
 
   connect():void {
-    this.toggleFolderDisplay(this.folderModeValue);
-    this.selectedFolderTextTarget.innerText = this.placeholderFolderNameValue;
-
-    const href = this.projectFolderHref;
-    if (href !== null) {
-      void fetch(href)
-        .then((data) => data.json())
-        .then((file:IStorageFile) => {
-          this.selectedFolderTextTarget.innerText = file.name;
-        });
-    }
+    combineLatest([
+      this.fetchStorageAuthorizationState(),
+      this.fetchProjectFolder(),
+    ]).subscribe(([isConnected, projectFolder]) => {
```

Comment:
Q: Curious why we need to wait for the latest values from authorization state and project folder- do we have [_"multiple long-lived observables"_](https://www.learnrxjs.io/learn-rxjs/operators/combination/combinelatest)? 🤔

Reply:
Well, it is true, that `combineLatest` has this use case, but the effect it has is perfect for this use case: it takes an array of observables and fires only if every single of them fired at least once.

`fetch` will return an observable, that fires exactly once. So, if you now read this line, it says:
1. Do 2 API fetch requests
2. Do something with the results ONLY IF both are completed.

Reply:
Nice one, cheers for the explanation 🎓 👍🏾